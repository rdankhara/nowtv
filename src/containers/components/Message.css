
li {
  border: 1px solid #ff9999;
  border-radius: 10px;
  list-style-type: none;
  margin-bottom: 10px;
  padding: 20px;
}

p {
  margin: 5px;
}
.flex-container {
  display: flex;
}

.avatar {
  width:100px;
  height: 100px;
  background-image: url('defaultAvatar.png');
  background-size: cover;
  margin-right: 10px;
}

.message-container{
  display: flex;
  flex-flow: column;
}

.message {
  margin:0px;
  height: 100px;
  background-color: beige;
  padding: 5px;
  border: 1px solid #555;
  border-radius: 5px;
}

.timestamp {
  align-self: flex-end;
}
.avatar:hover{
  cursor: pointer;
}


/* Tooltip container */
.tooltip {
  position: relative;
  display: inline-block;
  border-bottom: 1px dotted black; /* If you want dots under the hoverable text */
}

/* Tooltip text */
.tooltip .tooltiptext {
  visibility: hidden;
  width: 120px;
  background-color: #555;
  color: #fff;
  text-align: center;
  padding: 5px 0;
  border-radius: 6px;

  /* Position the tooltip text */
  position: absolute;
  z-index: 1;
  bottom: 125%;
  left: 50%;
  margin-left: -60px;

  /* Fade in tooltip */
  opacity: 0;
  transition: opacity 0.3s;
}

/* Tooltip arrow */
.tooltip .tooltiptext::after {
  content: "";
  position: absolute;
  top: 100%;
  left: 50%;
  margin-left: -5px;
  border-width: 5px;
  border-style: solid;
  border-color: #555 transparent transparent transparent;
}

/* Show the tooltip text when you mouse over the tooltip container */
.tooltip:hover .tooltiptext {
  visibility: visible;
  opacity: 1;
}

