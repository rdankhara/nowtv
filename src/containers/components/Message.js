import React from 'react';
import * as classes from './Message.css';
const Message = (props) => {
    return(
        <li>
            <article className="flexContainer">
                <a className="tooltip">
                    <img src={props.avatar} alt="" className="avatar tip" />
                    <span className="tooltiptext">{props.email}</span>
                </a>
                <div className="message-container">
                    <h2 className="fullName">{props.fullName}</h2>
                    { props.timestamp && 
                        <h5 className="timestamp">props.timestamp.toDateString()</h5>
                    }
                    <p className="message">{props.message}</p>
                </div>
            </article>
        </li>
    );
};

export default Message;