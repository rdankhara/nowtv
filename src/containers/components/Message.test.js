import React from 'react';
import Message from './Message';
import {shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';

configure({adapter: new Adapter()});

describe ('Message' , ()=>{
    let message ;
    
    beforeEach(()=> {
        message = shallow(<Message />);
    });

    it ('creates message component', ()=> {
        expect(message).toMatchSnapshot();
        expect(message.find('article')).not.toBeNull();
    });

    it('renders name correctly', ()=> {
        const props = {fullName :'Jack Richer'};
        message.setProps(props);

        const fullName = message.find('.fullName');
        expect(fullName).not.toBeNull();
        expect(fullName.text()).toBe('Jack Richer');
    });

    it('renders avatar correctly', ()=> {
        const avatar = 'http://dummyimage.com/100x100.jpg/dddddd/000000';
        const props = {avatar };
        message.setProps(props);

        const avatarEl = message.find('.avatar');
        expect(avatarEl).not.toBeNull();
        expect(avatarEl.prop('src')).toBe(avatar);
    });

    it('renders message correctly', ()=> {
        const msg = 'this is message';
        const props = {message : msg };
        message.setProps(props);

        const messageEl = message.find('.message');
        expect(messageEl).not.toBeNull();
        expect(messageEl.text()).toBe(msg);
    });    

    it('renders email correctly', ()=> {
        const email = 'abc@xyz.com';
        const props = {email};
        message.setProps(props);

        const emailEl = message.find('.tooltiptext');
        expect(emailEl).not.toBeNull();
        expect(emailEl.text()).toBe(email);
    });    

    it('renders timestamp correctly', ()=> {
        const timestamp = new Date('2017-02-09T04:27:38Z');
        const props = {timestamp};
        message.setProps(props);

        const timeEl = message.find('.timestamp');
        expect(timeEl).not.toBeNull();
        expect(timeEl.text()).toBe("Thu Feb 09 2017");
    });    
});