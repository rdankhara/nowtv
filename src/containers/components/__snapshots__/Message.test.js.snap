exports[`Message creates message component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Message />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <article
        className="flexContainer">
        <a
                className="tooltip">
                <img
                        alt=""
                        className="avatar tip"
                        src={undefined} />
                <span
                        className="tooltiptext" />
        </a>
        <div
                className="message-container">
                <h2
                        className="fullName" />
                <p
                        className="message" />
        </div>
</article>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <a
            className="tooltip">
            <img
                        alt=""
                        className="avatar tip"
                        src={undefined} />
            <span
                        className="tooltiptext" />
</a>,
          <div
            className="message-container">
            <h2
                        className="fullName" />
            <p
                        className="message" />
</div>,
        ],
        "className": "flexContainer",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt=""
                className="avatar tip"
                src={undefined} />,
              <span
                className="tooltiptext" />,
            ],
            "className": "tooltip",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "",
                "className": "avatar tip",
                "src": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "tooltiptext",
              },
              "ref": null,
              "rendered": null,
              "type": "span",
            },
          ],
          "type": "a",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2
                className="fullName" />,
              undefined,
              <p
                className="message" />,
            ],
            "className": "message-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "fullName",
              },
              "ref": null,
              "rendered": null,
              "type": "h2",
            },
            undefined,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "message",
              },
              "ref": null,
              "rendered": null,
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "article",
    },
    "type": "li",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <article
          className="flexContainer">
          <a
                    className="tooltip">
                    <img
                              alt=""
                              className="avatar tip"
                              src={undefined} />
                    <span
                              className="tooltiptext" />
          </a>
          <div
                    className="message-container">
                    <h2
                              className="fullName" />
                    <p
                              className="message" />
          </div>
</article>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <a
              className="tooltip">
              <img
                            alt=""
                            className="avatar tip"
                            src={undefined} />
              <span
                            className="tooltiptext" />
</a>,
            <div
              className="message-container">
              <h2
                            className="fullName" />
              <p
                            className="message" />
</div>,
          ],
          "className": "flexContainer",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <img
                  alt=""
                  className="avatar tip"
                  src={undefined} />,
                <span
                  className="tooltiptext" />,
              ],
              "className": "tooltip",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "",
                  "className": "avatar tip",
                  "src": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                  "className": "tooltiptext",
                },
                "ref": null,
                "rendered": null,
                "type": "span",
              },
            ],
            "type": "a",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h2
                  className="fullName" />,
                undefined,
                <p
                  className="message" />,
              ],
              "className": "message-container",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                  "className": "fullName",
                },
                "ref": null,
                "rendered": null,
                "type": "h2",
              },
              undefined,
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                  "className": "message",
                },
                "ref": null,
                "rendered": null,
                "type": "p",
              },
            ],
            "type": "div",
          },
        ],
        "type": "article",
      },
      "type": "li",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "prevContext": true,
          },
        },
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
